src/features/auth/hooks/validate-username-email.ts

```diff
@@ -8,26 +8,39 @@ const USERNAME_BLACKLIST = ['root'];
 
 for (const method of ['POST', 'PATCH'] as const) {
 	hooks.addPureHook(method, 'resin', 'user', {
-		POSTPARSE: async ({ request, tx }) => {
+		POSTPARSE: async ({ request }) => {
 			if (request.values.username != null) {
 				request.values.username = normalizeHandle(request.values.username);
 				if (USERNAME_BLACKLIST.includes(request.values.username)) {
 					throw new ConflictError('This username is blacklisted');
 				}
-
-				const existingUser = await findUser(request.values.username, tx, [
-					'id',
-				]);
-				if (existingUser) {
-					throw new ConflictError('This username is already taken');
-				}
-			}
-			if (request.values.email != null) {
-				const existingUser = await findUser(request.values.email, tx, ['id']);
-				if (existingUser) {
-					throw new ConflictError('This email is already taken');
-				}
 			}
 		},
+		PRERUN: async (args) => {
+			const { request, tx } = args;
+			await Promise.all(
+				(['username', 'email'] as const).map(async (field) => {
+					if (request.values[field] == null) {
+						return;
+					}
+					if (method === 'PATCH') {
+						const affectedIds = await sbvrUtils.getAffectedIds(args);
+						if (affectedIds.length === 0) {
+							return;
+						}
+						if (affectedIds.length > 1) {
+							throw new ConflictError(`The ${field} must be unique`);
```

Comment:
Can you put a `// TODO: This should be handled by a rule for case insensitive uniqueness` please